### PR TITLE
YLP-1256 Split profile page into several components

### PR DESCRIPTION
**Pull request by fatb38**

### CODE REVIEW

File: packages/yourloops/pages/profile/credentials-form.tsx [hunk 1]
```diff
@@ -69,12 +70,16 @@ function AuthenticationForm(props: AuthenticationFormProps): JSX.Element {
 
   return (
     <React.Fragment>
+      <div className={classes.categoryLabel}>
```
Comment (TimCanu) [inline packages/yourloops/pages/profile/credentials-form.tsx:73]: I'm not sure to understand this. This is a UI modification right?

File: packages/yourloops/pages/profile/profile.tsx [hunk 2]
```diff
@@ -161,50 +136,32 @@ const ProfilePage = (props: ProfilePageProps): JSX.Element => {
...
-      if (REGEX_BIRTHDATE.test(birthday)) {
-        setBirthDate(birthday);
-      } else {
-        setBirthDate("");
-      }
-    }
-    // No deps here, because we want the effect only when the component is mounting
-    // If we set the deps, the patient won't be able to change its birthday.
-  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+  const [firstName, setFirstName] = useState<string>(user.firstName);
```
Comment (TimCanu) [inline packages/yourloops/pages/profile/profile.tsx:139]: Suggestion: Having some many state does not ease code understandabilty and probably causes some excessive re-redeering. I propose slicing the states in objects (one object for each sub component). 
Contact me on slack if my comment is not clear!

File: packages/yourloops/pages/profile/profile.tsx [hunk 3]
```diff
@@ -342,162 +280,66 @@ const ProfilePage = (props: ProfilePageProps): JSX.Element => {
...
-              id="profile-button-save"
-              variant="contained"
-              disabled={!canSave}
-              color="primary"
-              onClick={onSave}
-              className={classes.button}
-            >
-              {t("button-save")}
-            </Button>
+            <ProgressIconButtonWrapper inProgress={saving}>
```
Comment (TimCanu) [inline packages/yourloops/pages/profile/profile.tsx:331]: This is new, right?

File: packages/yourloops/pages/profile/profile.tsx [hunk 4]
```diff
@@ -240,98 +197,79 @@ const ProfilePage = (props: ProfilePageProps): JSX.Element => {
...
+
       if (settingsChanged) {
         settings = await updateSettings(getUpdatedSettings(), false);
-        updated = true;
+        updatedUser.settings = settings;
       }
+
       if (role !== UserRoles.patient && passwordChanged) {
         await updatePassword(currentPassword, password);
+        setCurrentPassword("");
```
Comment (TimCanu) [inline packages/yourloops/pages/profile/profile.tsx:236]: Your code seems to be working without these three lines :thinking:

File: packages/yourloops/lib/auth/user.ts [hunk 5]
```diff
@@ -91,14 +91,14 @@ class User implements IUser {
   /**
    * Return the user first name
    */
-  getFirstName(): string {
+  get firstName(): string {
```
[the remark this answers is not in the document]
Comment (fatb38) [reply packages/yourloops/lib/auth/user.ts:94]: Class getters were currently not used, just improve the syntax and use it

File: packages/yourloops/pages/profile/credentials-form.tsx (hunk 1 above)
Comment (fatb38) [reply packages/yourloops/pages/profile/credentials-form.tsx:73]: moving the title into the components like the others

File: packages/yourloops/pages/profile/profile.tsx (hunk 3 above)
Comment (fatb38) [reply packages/yourloops/pages/profile/profile.tsx:331]: yeah, a progress loader. And also disable the button when making request, currently you can still click on it when saving

File: packages/yourloops/pages/profile/profile.tsx (hunk 4 above)
Comment (fatb38) [reply packages/yourloops/pages/profile/profile.tsx:236]: you're right it works ;)

File: packages/yourloops/pages/profile/profile.tsx (hunk 2 above)
Comment (fatb38) [reply packages/yourloops/pages/profile/profile.tsx:139]: For this case, even if each component has its own state, it's the Profile component that merge all states and call the hook, the problem will be the same and child components wild be re-rendered. But it's a good point for futur components ;)

**Review by TimCanu — APPROVED** [review]

**Pull request merged**